### ISSUE TITLE
49.3 : ATB, units and bots fail to load properly in the lobby

### ISSUE THREAD

**Issue by Question2** [MegaMek/mekhq#2899]
![image](https://user-images.githubusercontent.com/1191378/136338770-0e1eea5c-429f-49f8-9538-94b98d0752a6.png)

As an example, i started a battle, went into the lobby, but only my units were loaded. My ally was added as a bot, but their units did not load, and the bots for the OPFOR are missing.

This happens randomly and is not 100% replicable if you load the campaign save. Exiting the lobby and trying to start the battle usually does the trick, but this may take multiple tries for it to work.

This didnt happen in a previous version of mekhq (dont remember which version exactly though).

I did notice a lot of errors in the megameklog however : 
```

15:22:26,545 ERROR [mekhq.AtBGameThread] {My Campaign} 
run(), line 110 : Thread in unknown stage
```

Sometimes, when the units fail to load properly, mekhq gets stuck on 85% cpu usage and this lasts till i restart mekhq entirely.

[megameklog.txt](https://github.com/MegaMek/mekhq/files/7299779/megameklog.txt)

[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/7299780/mekhqlog.txt)

Campaign save : 

[b.zip](https://github.com/MegaMek/mekhq/files/7299802/b.zip)
[mechfiles.zip](https://github.com/MegaMek/mekhq/files/7299804/mechfiles.zip)

**Comment by Windchild292** [comment]
That logging just means it hasn't fully initialized, and I'll swap it to a warning.

Otherwise, this needs to be investigated to check if this has been resolved since 0.49.3

**Comment by Windchild292** [comment]
Subsumed by #2902, a duplicate issue from the same opener.
